crates/subspace-farmer/src/dsn/tests.rs

```diff
@@ -226,26 +217,31 @@ async fn test_dsn_sync() {
             last_archived_block.set_complete();
         }
     }
-    seeder_multiaddr.push({
-        schnorrkel::PublicKey::from_bytes(
-            &*seeder_multi_farming.single_plot_farms[0].plot.public_key(),
-        )
-        .map(identity::sr25519::PublicKey::from)
-        .map(identity::PublicKey::Sr25519)
-        .map(PeerId::from)
-        .map(Into::into)
-        .map(Protocol::P2p)
-        .unwrap()
-    });
 
-    let node_runners = futures::future::join_all(
-        std::mem::take(&mut seeder_multi_farming.networking_node_runners)
-            .into_iter()
-            .map(NodeRunner::run),
-    );
+    let (seeder_address_sender, mut seeder_address_receiver) = futures::channel::mpsc::unbounded();
+    seeder_multi_farming.single_plot_farms[0]
+        .node
+        .on_new_listener(Arc::new(move |address| {
+            seeder_address_sender
+                .unbounded_send(address.clone())
+                .unwrap();
```

Comment:
This will likely not work properly, it will panic when receiver is dropped. I see you copied code from example, code in example doesn't drop receiver, but it is still a bad code there.